### PR TITLE
[2452] Fix subject specialism mapping

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/OfcQm0PB/2452-fix-subject-specialism-mapping

When new subject mappings were added to DTTP, the "Refactor subject specialism mapping logic" PR was updated to reflect this. However, some subject like "Design and technology" which now have its own DTTP ID still need to show all the other subject specialisms.

### CODE REVIEW

**Comment by jhenderson** [comment]
[Fix subject specialism mapping](https://trello.com/c/OfcQm0PB/2452-fix-subject-specialism-mapping)

File: config/initializers/allocation_subjects.rb [hunk 1]
```diff
@@ -21,6 +21,6 @@ module AllocationSubjects
   PHYSICAL_EDUCATION = "Physical education"
   PHYSICS = "Physics"
   PRIMARY = "Primary"
-  RELIGIOUS_EDUCATION = "Religious education"
   PRIMARY_WITH_MATHEMATICS = "Primary with mathematics"
+  RELIGIOUS_EDUCATION = "Religious education"
```
Comment (jhenderson) [inline config/initializers/allocation_subjects.rb:25]: Ordering change

**Review by darokel — APPROVED** [review]

File: app/services/teacher_training_api/import_course.rb [hunk 2]
```diff
@@ -50,7 +49,13 @@ def invalid_age_range?
...
 
     def subjects
-      Subject.where(code: course_attributes[:subject_codes])
+      codes = course_attributes[:subject_codes]
+      # It's critical that the ordering of course.subjects ends up matching the order of course_attributes[:subject_codes]
+      # Why? Because the first subject will be mapped to an allocation subject and the allocation subject determines
+      # what bursary the trainee is entitled to. By ordering it correctly here, ActiveRecord will create the association
+      # records in the same order, and then we just need to add order("course_subjects.created_at") to Course#subjects.
+      # Adding an explicit "position" column to the course_subjects table was explored, but it's lot more complicated.
+      Subject.where(code: codes).sort_by { |subject| codes.index(subject.code) }
```
Comment (defong) [inline app/services/teacher_training_api/import_course.rb:58]: When restoring databases, will the order be ordered as expected?

Comment (jhenderson) [reply app/services/teacher_training_api/import_course.rb:60]: Yes, but the timestamps should be preserved, right?

Comment (defong) [reply app/services/teacher_training_api/import_course.rb:60]: Normally it's order it by `id`, as that's is the only guaranteed as atomic operation, as database is not guaranteed to returned in any order unless an order by clause is used.

**Review by edwardhorsford — APPROVED** [review]

**Pull request merged**